src/KDUtils/signal.h

```diff
@@ -0,0 +1,26 @@
+#ifndef KDUTILS_SIGNAL_H
+#define KDUTILS_SIGNAL_H
+
+#include <kdbindings/signal.h>
+
+using namespace KDBindings;
+
+namespace KDUtils {
+
+template<typename... Args>
+class Signal : public KDBindings::Signal<Args...>
+{
+public:
+    // New method to connect a slot through the CoreApplication's ConnectionEvaluator
+    ConnectionHandle connect(const std::shared_ptr<ConnectionEvaluator> &evaluator, std::function<void(Args...)> const &slot)
```

Comment:
Isn't this 1:1 the same API that `connectDeferred` has?

Why can't the user just call `connectDeferred`?
The rest of the PR looks fine :)